### ISSUE TITLE
segmentation fault clicking on Options

### ISSUE THREAD

**Issue by hongsudt** [informatics-isi-edu/deriva-qt#3]
When I first run deriva-py, I set up 3 servers (dev.rebuildingakidney.org, https://staging.rebuildingakidney.org, and www.rebuildingakidney.org).. Then I got `NoneType object has no attribute 'get'. ` 

Here is the stack trace: 
```
Traceback (most recent call last):

  File "/usr/bin/deriva-upload", line 9, in <module>
    load_entry_point('deriva.qt==0.4.1', 'console_scripts', 'deriva-upload')()

  File "/usr/lib/python3.5/site-packages/deriva/qt/upload_gui/__main__.py", line 12, in main
    sys.exit(gui.main())

  File "/usr/lib/python3.5/site-packages/deriva/qt/upload_gui/deriva_upload_gui.py", line 77, in main
    cookie_persistence=self.cookie_persistence)

  File "/usr/lib/python3.5/site-packages/deriva/qt/upload_gui/deriva_upload_gui.py", line 43, in upload_gui
    cookie_persistence=cookie_persistence)

  File "/usr/lib/python3.5/site-packages/deriva/qt/upload_gui/ui/upload_window.py", line 49, in __init__
    self.configure(uploader, hostname)

  File "/usr/lib/python3.5/site-packages/deriva/qt/upload_gui/ui/upload_window.py", line 80, in configure
    if not self.checkVersion():

  File "/usr/lib/python3.5/site-packages/deriva/qt/upload_gui/ui/upload_window.py", line 240, in checkVersion
    if not self.uploader.isVersionCompatible():

  File "/usr/lib/python3.5/site-packages/deriva/transfer/upload/deriva_upload.py", line 205, in isVersionCompatible
    compatibility = self.getVersionCompatibility()

  File "/usr/lib/python3.5/site-packages/deriva/transfer/upload/deriva_upload.py", line 202, in getVersionCompatibility
    return self.config.get("version_compatibility", list())

AttributeError: 'NoneType' object has no attribute 'get'
```

**Comment by robes** [comment]
The title and description of this issue probably need to be corrected because I believe this was a symptom of running the `deriva-upload` GUI with a ermrest catalog that lacked the `bulk-upload` annotation.

That said, the GUI should not crash and this is another issue with the initialization logic in `deriva.transfer.upload.deriva_upload.py` line 112 - 126 where it essentially bails out of the initialization process in a broken state. If this is intended to be terminal (which it basically is) then we should just throw an exception here and the in the higher level application code present the error to the user.

Also, the problem is coming from the deriva-py library so we might want to create a complementary issue there (or not).

**Comment by mikedarcy** [comment]
I've committed a fix for this to deriva-py: ba328d4. If you could test it out that would be great.  Please update both `deriva-py` and `deriva-qt`. You will need to delete/move your existing `.~/deriva` directory in order to reset things to the ground state that caused the bug.
